services/blockchain-service/blockchain-service-base.js

```diff
@@ -44,22 +40,24 @@ class BlockchainServiceBase {
 
     async prepareTransaction(contractInstance, functionName, args, blockchain) {
         const web3Instance = await this.getWeb3Instance(blockchain);
-        let gasLimit;
-        if (FIXED_GAS_LIMIT_METHODS[functionName]) {
-            gasLimit = FIXED_GAS_LIMIT_METHODS[functionName];
-        } else {
-            gasLimit = await contractInstance.methods[functionName](...args).estimateGas({
-                from: blockchain.publicKey,
-            });
-        }
+        const gasLimit = await contractInstance.methods[functionName](...args).estimateGas({
+            from: blockchain.publicKey,
+        });
 
         const encodedABI = await contractInstance.methods[functionName](...args).encodeABI();
 
-        let gasPrice;
+        let gasPrice = await web3Instance.eth.getGasPrice();
 
-        if (blockchain.name.startsWith('otp')) {
-            gasPrice = await web3Instance.eth.getGasPrice();
-        } else {
+        // Gas price increase for handling `Transaction not mined` error
+        if(blockchain.retryTx && gasPrice <= blockchain.gasPrice) {
+            gasPrice = Number(blockchain.gasPrice) + 1;
+        } else if (blockchain.gasPrice) {
+            // Gas price provided by developers
+            gasPrice = blockchain.gasPrice;
```

Comment:
Are we sure we want to use `blockchain.gasPrice` even if `web3Instance.eth.getGasPrice()` is bigger? If so, there is no point in getting `gasPrice` from the blockchain whatsoever, because it won't be used anyway as we're setting `blockchain.gasPrice` manually by default

Reply:
Well, if user for some reason wants with lower gas price I think we should let it use it.

Reply:
I agree with @u-hubar here, we should only get gas price from rpc node if blockchain.gasPrice is not defined

Reply:
Fixed